# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/tables/nsTableFrame.cpp
# Commit: de9e12276d8e
# Full Hash: de9e12276d8eac6610b71706a1b2c5035b924e5b
# Author: Ting-Yu Lin <tlin@mozilla.com>
# Date: 2022-10-14 21:55:00
# Description:
#   Bug 1795051 - Fix the condition before calling ResetColIndices in nsTableFrame::InsertColGroups. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159354
# ==============================================================================

diff -r 8afa6eaecd2f -r de9e12276d8e layout/tables/nsTableFrame.cpp
--- a/layout/tables/nsTableFrame.cpp	Fri Oct 14 16:03:35 2022 +0000
+++ b/layout/tables/nsTableFrame.cpp	Fri Oct 14 16:31:47 2022 +0000
@@ -529,7 +529,7 @@
     colIndex += numCols;
   }
 
-  for (; *colGroupIter; ++colGroupIter) {
+  if (*colGroupIter) {
     nsTableColGroupFrame::ResetColIndices(*colGroupIter, colIndex);
   }
 }
